cypress/integration/Switch.spec.ts

```diff
@@ -0,0 +1,69 @@
+import * as h from '../helpers';
+
+const getSwitch = () => {
+  return cy.get(`[type="checkbox"]`);
+};
+
+describe('Non-disabled Switch', () => {
+  before(() => {
+    h.stories.visit();
+  });
+  ['Default', 'Alert', 'Error'].forEach(story => {
+    context(`Given the '${story}' story is rendered`, () => {
+      beforeEach(() => {
+        h.stories.load('Components|Inputs/Switch/React/Top Label', story);
+      });
+
+      it('should not have any axe errors', () => {
+        cy.checkA11y();
+      });
+
+      context('when the tab key is pressed', () => {
+        it('should be the focused item on the page', () => {
+          cy.tab();
+          getSwitch().should('have.focus');
+        });
+      });
+
+      context('when the switch is clicked', () => {
+        it('should be toggled', () => {
+          getSwitch()
+            .invoke('prop', 'checked')
+            .as('checked');
+          getSwitch().click();
+          getSwitch().should(this.checked ? 'not.be.checked' : 'be.checked');
```

Comment:
I don't think it is a good idea to have conditional testing. Either we expect a checkbox to be checked after a click or we expect it to not be checked after a click

Reply:
If you expect the default loaded state to be unchecked, have a spec for that. Then this spec clicks the switch and verifies it is now checked